lib/hardware_services/inventory_drivers/QuadsNativeInventoryDriver.py

```diff
@@ -153,12 +153,45 @@ def load_data(self, quadsinstance, force):
             quadsinstance.logger.error(ex)
             exit(1)
 
-    def write_data(self, quadsinstance, doexit = True):
-        quadsinstance.quads_write_data(doexit)
+    def write_data(self, quadsinstance, doexit):
+        try:
+            stream = open(quadsinstance.config, 'w')
+            quadsinstance.data = {"clouds":quadsinstance.quads.clouds.data,
+             "hosts":quadsinstance.quads.hosts.data, "history":quadsinstance.quads.history.data, "cloud_history":quadsinstance.quads.cloud_history.data}
+            stream.write( yaml.dump(quadsinstance.data, default_flow_style=False))
+            if doexit:
+                exit(0)
+        except Exception, ex:
+            quadsinstance.logger.error("There was a problem with your file %s" % ex)
```

Comment:
You should let the logging function do the formatting instead of using python's `%` operator. 

so instead of this
```python
logger.error('Foo: %r' % bar)
```
we should do 
```python
logger.error('Foo: %r', bar)
```

for more details, have a look at this issue https://github.com/CCI-MOC/hil/issues/629